src/appleseed/renderer/modeling/project/projectfilereader.cpp

```diff
@@ -3036,15 +3038,54 @@ namespace
     }
 }
 
+bool is_packed_project(const string& project_filepath, string& project_name) 
+{
+    try 
+    {
+        vector<string> appleseed_files = get_filenames_with_extension_from_zip(project_filepath, ".appleseed");
+        
+        if (appleseed_files.size() != 1) {
+            RENDERER_LOG_INFO(
+                "%s could be a packed project, but only one .appleseed file should be inside.", 
+                project_filepath.c_str());
+            return false;
+        }
+
+        project_name = appleseed_files[0];
+        return true;
+    }
+    catch (UnzipException e)
+    {
+        return false;
+    }
+}
+
 auto_release_ptr<Project> ProjectFileReader::read(
     const char*             project_filepath,
     const char*             schema_filepath,
     const int               options)
 {
     assert(project_filepath);
+    
+    string project_name;
+
+    // Handle packed projects
+    string actual_project_filepath;
+    if (is_packed_project(project_filepath, project_name))
+    {
+        string unpacked_project_directory(project_filepath);
+        unpacked_project_directory += ".unpacked";
+
+        RENDERER_LOG_INFO("%s appears to be a packed project.", project_filepath);
+        RENDERER_LOG_INFO("unpacking to %s...", unpacked_project_directory.c_str());
```

Comment:
I suggest combining these two messages:

    RENDERER_LOG_INFO("%s appears to be a packed project; unpacking to %s...", project_filepath, unpacked_project_directory.c_str());